docs/publish.py

```diff
@@ -1,41 +1,76 @@
 #!/usr/bin/python3
 import os
 import sys
-__dir__ = os.path.dirname(__file__)
-sys.path.insert(0, os.path.join(__dir__, ".."))
+import typing as ty
+
+script_dir = os.path.dirname(os.path.realpath(__file__))
+sys.path.insert(0, os.path.join(script_dir, ".."))
```

Comment:
IMHO its almost always nicer to use `pathlib.Path`, rather then `os.path` when dealing with filesystem paths:

```py
__dirpath__ = pathlib.Path(__file__).resolve().parent
sys.path.insert(0, __dirpath__ / "..")
```

(Just a suggestion though.)

Reply:
`pathlib.Path` appeals to me also, I was maintaining the status quo here.